src/main/java/cpp/storage/Storage.java

```diff
@@ -49,28 +49,30 @@ private static ProjectList readData() throws IOException {
             //System.out.println("Welcome new user!");
             createFile("./src/data");
         }
+        System.out.println("---- " + new File(".").getAbsolutePath());
         BufferedReader in = new BufferedReader(new FileReader(".\\src\\data\\projectList.txt"));
         String projectLine;
         ProjectList projectList = new ProjectList();;
         int indexProject = 1;
         while ((projectLine = in.readLine()) != null) {
-            String[] details = projectLine.split(";");
+            String[] details = projectLine.split("`");
             int indexTodo = 1;
-            assert (details.length == 3) : "Unable to load data! Data is incomplete!";
+            assert (details.length == 4) : "Unable to load data! Data is incomplete!";
 
             String title = details[0];
             String todos = details[1];
             String deadline = details[2];
+            final String gitHubLink = details[3];
 
             //add project to list
             projectList.addProject(title);
 
             String[] todoInfo = todos.split(",");
             //add todo to project
-            for (int i = 0; i < (todoInfo.length) / 2; i++) {
-                String todoDescrip = todoInfo[2 * i];
-                String todoStatus = todoInfo[2 * i + 1];
-                String todoDeadline = todoInfo[2 * i + 2];
+            for (int i = 0; i < (todoInfo.length - 1) / 2; i++) {
```

Comment:
It should not be `(todoInfo.length-1)/2`. If there are three elements saved in each todo, it should be `(todoInfo.length)/3`.

Reply:
Fixed 👍